File: Microsoft .NET/.NET Framework 4/ASP.NET/VB.NET/MVC Web Reporting Sample/MVC Web Reporting Sample/Controllers/LLWebReportViewerController.vb
Imports combit.Reporting
Imports combit.Reporting.Web.WebReportViewer
Imports System.IO

Namespace WebReporting.Controllers
    Public Class LLWebReportViewerController
        Inherits WebReportViewerController

        Public Overrides Sub OnProvideListLabel(ByVal provideListLabelContext As ProvideListLabelContext)
            Dim repositoryIdOfProject As String = provideListLabelContext.ProjectName
            Dim ll As ListLabel = DefaultSettings.GetListLabelInstance(repositoryIdOfProject, DefaultSettings.GetBaseRepository())
            ll.Core.LlXSetParameter(LlExtensionType.Export, "HTML5", "XHTML.ForceTableLineOneHundredPercentageWidth", "1")
            provideListLabelContext.ExportPath = Path.Combine(SampleWebReportingApplication.TempDirectory, Guid.NewGuid().ToString("D"))
            ll.Variables.Add("ItemList.ItemNo.From", String.Empty)
            ll.Variables.Add("ItemList.ItemNo.To", String.Empty)
            provideListLabelContext.NewInstance = ll
        End Sub

        Public Overrides Sub OnProvideRepository(ByVal provideFileRepositoryContext As ProvideRepositoryContext)
            provideFileRepositoryContext.FileRepository = DefaultSettings.GetBaseRepository()
        End Sub
    End Class
End Namespace

File: Microsoft .NET/.NET Framework 4/ASP.NET/C#/MVC Web Reporting Sample/MVC Web Reporting Sample/Controllers/LLWebReportViewerController.cs
using combit.Reporting;
using combit.Reporting.Web.WebReportViewer;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Web;
using System.Web.Mvc;

namespace WebReporting.Controllers
{
    public class LLWebReportViewerController : WebReportViewerController
    {
        public override void OnProvideListLabel(ProvideListLabelContext provideListLabelContext)
        {
            string repositoryIdOfProject = provideListLabelContext.ProjectName;
            ListLabel ll = DefaultSettings.GetListLabelInstance(repositoryIdOfProject, DefaultSettings.GetBaseRepository());

            ll.Core.LlXSetParameter(LlExtensionType.Export, "HTML5", "XHTML.ForceTableLineOneHundredPercentageWidth", "1");
            // D:   Der WebReportViewer benötigt ein Verzeichnis für temporäre Dateien. Diese werden einige Minuten nach Schließen eines WebReportViewer automatisch gelöscht.
            // US:  The WebReportViewer requires a directory for temporary files. Some minutes after a WebReportViewer is closed, these files will be deleted automatically.
            provideListLabelContext.ExportPath = SampleWebReportingApplication.TempDirectory; //use the same tempDir (Demo)
			
            //LL.Variables.Add("ArtikelListe.ArtikelNr.Von", string.Empty);
            //LL.Variables.Add("ArtikelListe.ArtikelNr.Bis", string.Empty);

            ll.Variables.Add("ItemList.ItemNo.From", string.Empty);
            ll.Variables.Add("ItemList.ItemNo.To", string.Empty);

            provideListLabelContext.NewInstance = ll;
        }

        public override void OnProvideRepository(ProvideRepositoryContext provideFileRepositoryContext)
        {
            provideFileRepositoryContext.FileRepository = DefaultSettings.GetBaseRepository();
        }

    }
}